lib/socket.c

```diff
@@ -303,7 +303,7 @@ smb2_write_to_socket(struct smb2_context *smb2)
         return 0;
 }
 
-typedef ssize_t (*read_func)(struct smb2_context *smb2,
+typedef size_t (*read_func)(struct smb2_context *smb2,
```

Comment:
I don't think we can do this change from ssize_t to size_t.
The helper functions are based on how functions like readv() returns error (<0)
and smb2_read_data() depends on this.
Line 351 in socket.c , after we hare read the data via these functions, smb2_read_data() depends on them returning <0.

Leave these function returning ssize_t for now, that way the mimic posix functiions for nonblocking i/o. return >0 for number of bytes read, <0 for check errno, buffer empty and try again later   or ==0 for end of file

Reply:
Ok, i will leave it, btw i changed also some variables that takes size_t instead of ssize_t.